test/dynamic_realm_test.dart

```diff
@@ -0,0 +1,197 @@
+      expect(allCars.length, 1);
+
+      final car = allCars[0];
+      expect(RealmObject.get<String>(car, 'make'), 'Honda');
+    });
+
+    test('dynamic.all (dynamic=$isDynamic) throws for non-existent type', () {
+      final config = Configuration.local([Car.schema]);
+      final staticRealm = getRealm(config);
+
+      final dynamicRealm = getDynamicRealm(staticRealm);
+
+      expect(() => dynamicRealm.dynamic.all('i-dont-exist'), throws<RealmError>("Object type i-dont-exist not configured in the current Realm's schema"));
+    });
+
+    test('dynamic.all (dynamic=$isDynamic) can follow links', () {
+      final config = Configuration.local([LinksClass.schema]);
+      final staticRealm = getRealm(config);
+
+      final id1 = Uuid.v4();
+      final id2 = Uuid.v4();
+      final id3 = Uuid.v4();
+
+      staticRealm.write(() {
+        final obj1 = staticRealm.add(LinksClass(id1));
+        final obj2 = staticRealm.add(LinksClass(id2));
+        final obj3 = staticRealm.add(LinksClass(id3));
+
+        obj1.link = obj2;
+        obj2.link = obj3;
+
+        obj1.list.addAll([obj1, obj2, obj3]);
+      });
+
+      final dynamicRealm = getDynamicRealm(staticRealm);
+
+      final objects = dynamicRealm.dynamic.all(LinksClass.schema.name);
+      final obj1 = objects.singleWhere((o) => RealmObject.get<Uuid>(o, 'id') as Uuid == id1);
+      final obj2 = objects.singleWhere((o) => RealmObject.get<Uuid>(o, 'id') as Uuid == id2);
+      final obj3 = objects.singleWhere((o) => RealmObject.get<Uuid>(o, 'id') as Uuid == id3);
+
+      expect(RealmObject.get<RealmObject>(obj1, 'link'), obj2);
+      expect(RealmObject.get<RealmObject>(obj2, 'link'), obj3);
+
+      final list = RealmObject.get<RealmObject>(obj1, 'list') as List<RealmObject>;
+
+      expect(list[0], obj1);
+      expect(list[1], obj2);
+      expect(list[2], obj3);
+    });
+
+    test('dynamic.all (dynamic=$isDynamic) can be filtered', () {
+      final config = Configuration.local([Car.schema]);
+      final staticRealm = getRealm(config);
+
+      staticRealm.write(() {
+        staticRealm.add(Car('Honda'));
+        staticRealm.add(Car('Hyundai'));
+        staticRealm.add(Car('Suzuki'));
+        staticRealm.add(Car('Toyota'));
+      });
+
+      final dynamicRealm = getDynamicRealm(staticRealm);
+
+      final carsWithH = dynamicRealm.dynamic.all(Car.schema.name).query('make BEGINSWITH "H"');
+      expect(carsWithH.length, 2);
+    });
+
+    test('dynamic.find (dynamic=$isDynamic) can find by primary key', () {
+      final config = Configuration.local([Car.schema]);
+      final staticRealm = getRealm(config);
+
+      staticRealm.write(() {
+        staticRealm.add(Car('Honda'));
+        staticRealm.add(Car('Hyundai'));
+      });
+
+      final dynamicRealm = getDynamicRealm(staticRealm);
+
+      final car = dynamicRealm.dynamic.find(Car.schema.name, 'Honda');
+      expect(car, isNotNull);
+      expect(RealmObject.get<String>(car!, 'make'), 'Honda');
```

Comment:
this will be 
```
car.make
```
if we overwrite `noSuchMethod` in `DynamicRealmObject`

Reply:
As mentioned in the other comment, this is done in another PR: https://github.com/realm/realm-dart/pull/524/files#diff-c2ba1d881cf423a37576e2f5fd6b75b73bac7783f002a36eab9cb57815d4290dR289.

Reply:
resolving